### PR TITLE
Fix CI 

### PR DESCRIPTION
**Pull request by tcharding**
Fix CI by doing:

- Pin `memchr`
- Fix new clippy version warnings (remove unnecessary `vec`)

### CODE REVIEW

**Review by apoelstra — APPROVED** [review]
ACK 5334b2177c45247635a8a7b3e7f6953ec067658d

File: contrib/test.sh [hunk 1]
```diff
@@ -13,6 +14,11 @@ if cargo --version | grep nightly >/dev/null; then
     NIGHTLY=true
 fi
 
+if cargo --version | grep ${MSRV}; then
+    # memcrh 2.6.0 uses edition 2021
```
Comment (tcharding) [inline contrib/test.sh:18]: fix this, like in https://github.com/rust-bitcoin/hex-conservative/pull/33/files

**Comment by tcharding** [comment]
Fixed the memchr typo in the docs, no other changes.

**Review by apoelstra — APPROVED** [review]
ACK cce557474f64fe0f4d3aa9df18af245c1e82f427

**Pull request merged**